include/saber/inexact.hpp

```diff
@@ -0,0 +1,71 @@
+#ifndef SABER_INEXACT_HPP
+#define SABER_INEXACT_HPP
+#pragma once
+
+// std
+#include <algorithm>
+#include <cmath>
+#include <limits>
+#include <cassert>
+#include <type_traits>
+
+namespace saber {
+
+class Inexact
+{
+public:
+    template<typename T>
+    static bool IsEQ(T inLHS, T inRHS)
+    {
+        EQ equal{inLHS};
+        const bool result = equal(inRHS);
+        return result;
+    }
+
+    /*
+    template<typename T>
+    static bool IsNE(T inLHS, T inRHS)
+    {
+        EQ unequal{inLHS};
+        const bool result = unequal(inRHS);
+        return result;
+    }
+    */
+
+    template<typename T>
+    struct EQ
+    {
+    public:
+        EQ(const T& inLHS) :
+            mLHS{inLHS}
+        {
+            bool assertion = std::is_floating_point<T>::value;
```

Comment:
You will never change the value of this `bool`.
Ordinarily, you would say, "oh, right... `const`".
But the value of `is_floating_point` for a given type is compile-time-known, so:
1. `constexpr`
2. `assertion` doesn't convey information about the value the variable is holding. Consider using: `isFloatingPoint`. You'll want the code to read: `static_assert(isFloatingPoint)`, instead of `static_assert(assertion)`. Because the former provides superior intent than the latter.
3. Subjective: STL provides `std::is_XXXX_v<T>` synonyms as shorthand for `std::is_XXX<T>::value`